public/js/wyr-single.js

```diff
@@ -190,3 +135,97 @@ async function createComment() {
     console.log(err);
   }
 }
+
+/* ADDING AND DELETING QUESTION UI */
+const addQuestionBtn = document.getElementById("add-question-button");
+const questionModal = document.getElementById("add-question-modal");
+const questionForm = document.getElementById("addQuestionForm");
+const addQuestionCloseModal = document.getElementById(
+  "addQuestion-close-modal"
+);
+const deleteQuestionBtn = document.getElementById("deleteQuestionBtn");
+
+addQuestionBtn.addEventListener("click", () => {
+  questionModal.classList.add("show");
+  questionModal.classList.remove("hidden");
+});
+
+addQuestionCloseModal.addEventListener("click", () => {
+  questionModal.classList.add("hidden");
+  questionModal.classList.remove("show");
+});
+
+questionForm.addEventListener("submit", (e) => {
+  e.preventDefault();
+  addNewQuestion();
+});
+
+async function addNewQuestion() {
+  const newOption1 = document.getElementById("newOption1").value;
+  const newOption2 = document.getElementById("newOption2").value;
+
+  const userId = document.getElementById("user-id").value;
+
+
+  try {
+    const response = await fetch('/questions', {
+      method: 'post',
+      headers: { 'Content-type': 'application/json' },
+      body: JSON.stringify({
+        option1: newOption1,
+        option2: newOption2,
+        voteCount1: 0,
+        voteCount2: 0,
+        createdBy: userId,
+      })
+    });
+    const data = await response.json();
+    console.log(data);
+
+    const questionList = document.getElementById("questionList");
+    const newQuestion = document.createElement("li")
+    const novelOption1 = document.createElement("p");
+    const novelOption2 = document.createElement("p");
+    const newQuestionUser = document.createElement("p");
+    const deleteBtn = document.createElement("button");
+    newQuestion.classList.add("flex","flex-row","items-start","w-[100%]","border-b","border-gray-500","mb-2");
+    newQuestion.setAttribute("data-id", data._id);
+    newQuestionUser.classList.add("text-gray-600", "break-words", "w-[24%]");
+    newQuestionUser.textContent = data.createdBy;
+    novelOption1.classList.add("text-gray-600", "break-words", "w-[24%]");
+    novelOption1.textContent = data.option1;
+    novelOption2.classList.add("text-gray-600", "break-words", "w-[24%]");
+    novelOption2.textContent = data.option2;
+    newQuestion.appendChild(novelOption1);
+    newQuestion.appendChild(novelOption2);
+    if (data.createdBy === userId) {
+      deleteBtn.textContent = "🗑️";
+      deleteBtn.classList.add("w-[24%]");
+      deleteBtn.id = "deleteQuestionBtn";
+      deleteBtn.addEventListener("click", deleteQuestion);
+      newQuestion.appendChild(deleteBtn);
+    }
+    console.log("newQuestion", newQuestion );
+    questionList.appendChild(newQuestion);
+    newOption1.value = "";
+    newOption2.value = "";
+  } catch (err) {
+    console.log(err);
+  }
+}
+
```

Comment:
above deals with adding the question. Hopefully when the controller functions for adding the question is added, this should work as intended. It adds the question to the top of the modal, just like in the addingComments modal